internal/testrunner/runners/system/system.go

```diff
@@ -2,30 +2,134 @@ package system
-func Run(testFolderPath string) error {
-	r := runner{testFolderPath}
+func Run(testFolder testrunner.TestFolder, packageRootPath string) error {
+	r := runner{
+		testFolder,
+		packageRootPath,
+		getStackSettingsFromEnv(),
+	}
 	return r.run()
 }
 
 func (r *runner) run() error {
-	fmt.Println("system run", r.testFolderPath)
+	// Step 1. Setup service.
+	// Step 1a. (Deferred) Tear down service.
+	logger.Info("Setting up service...")
+	serviceRunner, err := servicedeployer.Factory(r.packageRootPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create service runner")
+	}
+
+	ctxt := common.MapStr{}
+	ctxt, err = serviceRunner.SetUp(ctxt)
+	defer func() {
+		logger.Info("tearing down service...")
+		if err := serviceRunner.TearDown(ctxt); err != nil {
+			logger.Errorf("error tearing down service: %s", err)
+		}
+	}()
+	if err != nil {
+		return errors.Wrap(err, "could not setup service")
+	}
+
+	// Step 2. Setup agent and enroll it with Fleet
+	// TODO: Should this be part of the service factory, since the agent might need
+	// to be deployed "close to" the service? So for docker-compose based services,
+	// we could use the Agent Docker image, for TF-based services, we use TF to
+	// deploy the agent "near" the service, etc.?
+	// TODO: mount ctxt.Get("stdoutFilePath") and ctxt.Get("stderrFilePath")
+
+	// Step 3. Configure package (single data stream) via Ingest Manager APIs.
+	im, err := ingestmanager.NewClient(r.stackSettings.kibana.host, r.stackSettings.elasticsearch.username, r.stackSettings.elasticsearch.password)
+	if err != nil {
+		return errors.Wrap(err, "could not create ingest manager client")
+	}
+
+	p := ingestmanager.Policy{
+		Name:        fmt.Sprintf("ep-test-system-%s-%s", r.testFolder.Package, r.testFolder.Dataset),
+		Description: fmt.Sprintf("test policy created by elastic-package test system for data stream %s/%s", r.testFolder.Package, r.testFolder.Dataset),
+		Namespace:   "ep",
+	}
+	policy, err := im.CreatePolicy(p)
+	if err != nil {
+		return errors.Wrap(err, "could not create policy")
+	}
+	defer func() {
+		time.Sleep(5 * time.Minute) // TODO: remove
+		if err := im.DeletePolicy(*policy); err != nil {
+			logger.Errorf("error cleaning up policy: %s", err)
+		}
+	}()
+
+	// TODO: build data stream config by taking appropriate vars sections from package manifest + dataset manifest,
+	// starting with defaults, then overridding with vars from {dataset}/_dev/test/system/vars.yml. Then treat result
+	// as go template and evaulate against ctxt. See expected final structure in im.AddPackageToPolicy method.
+
+	// TODO: add package/datastream config to policy (im.AddPackageToPolicy)
+
+	fmt.Println(policy)
+
+	// Step 4. (TODO in future) Optionally exercise service to generate load.
+
+	// Step 5. Assert that there's expected data in data stream.
+
 	return nil
 }
 
-func init() {
-	testrunner.RegisterRunner(TestType, Run)
+func getStackSettingsFromEnv() stackSettings {
```

Comment:
I exposed environment variables here: https://github.com/elastic/elastic-package/blob/c3fd692e9923f4160b11b3eac569b76014552934/internal/elasticsearch/client.go Does it make sense to use them here?

Reply:
I'm open to it but I'm not sure how to do that. Can you elaborate what should change here?

Reply:
I suppose it would be merging the other PR and adjust the implementation here to use environment variables defined in a single place (e.g. close to the stack).

Reply:
Okay, I plan to bring the changes from the other PR into this one today. Let's see how things look at that point.